components/list/list-item.js

```diff
@@ -31,327 +8,33 @@ const defaultBreakpoints = [842, 636, 580, 0];
  * @slot actions - Actions (e.g., button icons) associated with the listen item located at the right of the item
  * @fires d2l-list-item-selected - Dispatched when the component item is selected
  */
-class ListItem extends ListItemDragDropMixin(ListItemCheckboxMixin(ListItemRoleMixin(RtlMixin(LitElement)))) {
-
+class ListItem extends ListItemMixin(LitElement) {
 	static get properties() {
 		return {
-			/**
-			 * Breakpoints for responsiveness in pixels. There are four different breakpoints and only the four largest breakpoints will be used.
-			 */
-			breakpoints: { type: Array },
 			/**
 			 * Address of item link if navigable
 			 */
-			href: { type: String },
-			_breakpoint: { type: Number },
-			_dropdownOpen: { type: Boolean, attribute: '_dropdown-open', reflect: true },
-			_hoveringLink: { type: Boolean },
-			_focusing: { type: Boolean },
-			_focusingLink: { type: Boolean },
-			_tooltipShowing: { type: Boolean, attribute: '_tooltip-showing', reflect: true }
+			action: { type: String, reflect: true },
```

Comment:
It looks like this new `action` property is already on the `ListItemMixin` - just need to update it to reflect.

From what I gather, the `href` on the item (for navigation) potentially conflicts with the `href` for a hypermedia-drive component - yes?  One might argue that these `href`s ought to be the same.  Food for thought.  🤔 

Before going much further with this though, I think it is worthwhile settling on our solution for the JS action scenario... I recall we discussed this a bit a few weeks ago.  To preface, we will need the ability to define the primary action to be either JS (button semantics) or navigation (anchor semantics), while we currently only support the navigation scenario.  It would be ideal if the consumer could provide the element, leaving the list-item un-opinionated and non-limiting.  For instance, currently we're only exposing `href`, but there are other link specific properties we are omitting, such as `target`.  Otherwise, the consumer needs to tell the list item what type of action it is, so that it can render an `a` or `button`.  I think we talked about the button dispatching a custom event that the consumer could listen for.  In short, let's make sure any API changes we make jive with that use-case.

Reply:
>From what I gather, the href on the item (for navigation) potentially conflicts with the href for a hypermedia-drive component - yes? One might argue that these hrefs ought to be the same. Food for thought.

I would really like them to be the same but are used differently. One links to a webpage where the other links to an api. If you have thoughts on how we can treat them as the same I would love to hear that when you get back.

>It looks like this new action property is already on the ListItemMixin - just need to update it to reflect.

I am reflecting in this case so `href` is backwards compatible. Since all other cases don't need it reflected I decided to just add the special case on the `list-item` component.

Yeah, I decided in another PR to handle the event. I believe we agreed that people can either set `action` as a url or define a click handler using a custom event.

I would also prefer not to use name `action` as the url attribute. If anyone can think of a better name I'm all ears.

Reply:
What about `actionHref` / `action-href`?  That's the approach that was taken with the MVC DataList... it uses `ActionLocation` (`ILocation`) and `ActionOnClick` (`IJavaScriptFunction`).

Reply:
Awesome!